lib/sinon/util/fake_server.js

```diff
@@ -74,8 +74,10 @@ var fakeServer = {
             this.xhr = sinon.useFakeXMLHttpRequest();
         }
         server.requests = [];
-
         this.xhr.onCreate = function (xhrObj) {
+            xhrObj.unsafeHeadersEnabled = function () {
```

Comment:
With function approach it is a much cleaner approach as the property value is maintained in one single place - server. Otherwise you introduce a disconnect between the server and http request for no reason. What if decide to change the value of this property at server level during the lifecycle of the app? If I propagate the property value down to the http request level, the request will never see the new server level value but rather use the now invalid original property?

Makes sense ?